apps/condo/domains/organization/utils/accessSchema.js

```diff
@@ -1,239 +1,308 @@
+            organizationIds.push(...info.childOrganizations)
+        }
     }
-
-    return true
+    return uniq(organizationIds)
 }
 
-async function checkUserBelongsToOrganization (userId, organizationId) {
-    if (!userId || !organizationId) return false
-    const employee = await getByCondition('OrganizationEmployee', {
-        organization: { id: organizationId },
-        user: { id: userId },
-        isAccepted: true,
-        isBlocked: false,
-        isRejected: false,
-        deletedAt: null,
-    })
+/**
+ * Gets the IDs of organizations and its child (related) organizations in whose parent organizations the user is employed and has all permissions from the list.
+ * @param {{ id: string }} user - user object
+ * @param {Array<string> | string} permissions - permissions to check,
+ * can be passed as array of strings for multiple permissions or a single string for a single permission
+ * @returns {Promise<Array<string>>}
+ */
+async function getEmployedOrRelatedOrganizationsByPermissions (user, permissions) {
+    const userOrganizationsInfo = await _getUserOrganizations(user)
 
-    if (!employee) {
-        return false
-    }
+    const permissionsToCheck = Array.isArray(permissions) ? permissions : [permissions]
+    const organizationIds = []
 
-    if (employee.isBlocked) {
-        return false
+    for (const [id, info] of Object.entries(userOrganizationsInfo.organizations)) {
+        if (_checkPermissionsInOrganization(info.permissions, permissionsToCheck)) {
+            organizationIds.push(id, ...info.childOrganizations)
+        }
     }
+    return uniq(organizationIds)
+}
 
-    return employee.deletedAt === null
+/**
+ * Checks if user is employed in all listed organizations and has all correct permissions in it.
+ * Both organizations and permissions can be single elements if passed as strings instead of arrays
+ * This utils is faster than filtering organization ids from corresponding get<> function,
+ * so use it when you don't need related organizations
+ * @param {{ id: string }} user - user object
+ * @param {Array<string> | string} organizationIds - organizations to checks (can be passed as array of IDs or a single ID)
+ * @param {Array<string> | string} permissions - permissions to check (can be passed as array or a single string)
+ * @returns {Promise<boolean>}
+ */
+async function checkPermissionsInEmployedOrganizations (user, organizationIds, permissions) {
+    const userOrganizationsInfo = await _getUserOrganizations(user)
+
+    const organizationsToCheck = Array.isArray(organizationIds) ? uniq(organizationIds) : [organizationIds]
+    const permissionsToCheck = Array.isArray(permissions) ? permissions : [permissions]
+
+    return organizationsToCheck.every(orgId =>
+        userOrganizationsInfo.organizations.hasOwnProperty(orgId) &&
+        _checkPermissionsInOrganization(
+            userOrganizationsInfo.organizations[orgId].permissions,
+            permissionsToCheck
+        )
+    )
 }
```

Comment:
Сheck that `organizationsToCheck`/`organizationIds` is not empty.
It is important. If there is an empty array, then access will be allowed (`[].every(cb)` return true); this should not happen. He should be banned

Reply:
Why? It was unclear behaviour before :)

Why do you assume that empty array is a bad thing? 
For example I can do request like this:
```graphql
mutation {
    createSomething (data: {
        relation: { create: { ...relationPayload } }
    }) {
        id
    }
}
```

So your typical `get(originalInput, 'relation.connect.id')` will be null, and `[]` is valid to resolve in true, since I do not connect any ogranization

Reply:
If you do not use an organization when checking access, then most likely you do not need these utilities)

Reply:
You want the developer to check that he is not passing in an empty organization(s). But the developer will expect that the utility will check everything itself))